### PR TITLE
Changes to /debug endpoints for InfluxDB 2.0.5

### LINKED ISSUE #2239: InfluxDB 2.0.5: Changes to /debug endpoints
**PR:** influxdata/influxdb#20860

* A new `--pprof-disabled` flag to `influxd` allows users to disable all `/debug/pprof` endpoints
* `influxd upgrade` has been extended to convert the 1.x `http.pprof-enabled` setting into an equivalent 2.x `pprof-disabled` setting in config
* When enabled, a new `/debug/pprof/all` endpoint allows for collecting all profiling information in a single request
* The `/debug/vars` endpoint has been removed. The `/metrics` endpoint is/was meant to replace it in 2.x, but it suck into our previous builds because some dependency was enabling it.

### PR DESCRIPTION
**Pull request by pierwill**
- Add --pprof-disabled influxd flag
- Add pprof-disabled to influxd config-options

Closes #2239.

- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

File: content/influxdb/v2.0/reference/config-options.md [hunk 1]
```diff
@@ -821,6 +821,53 @@ no-tasks = true
 
 ---
 
+### pprof-disabled
+Don't expose debugging information over HTTP at `/debug/pprof`.
```
Comment (sanderson) [inline content/influxdb/v2.0/reference/config-options.md:825]:
```suggestion
Disable the `/debug/pprof`  HTTP endpoint. This endpoint provides runtime profiling data and can be helpful when debugging.
```

File: content/influxdb/v2.0/reference/config-options.md [hunk 2]
```diff
@@ -821,6 +821,53 @@ no-tasks = true
 
 ---
 
+### pprof-disabled
```
Comment (sanderson) [inline content/influxdb/v2.0/reference/config-options.md:824]: Be sure to add a link to this anchor in the TOC near the top of the page.

File: content/influxdb/v2.0/reference/config-options.md [hunk 3]
```diff
@@ -821,6 +821,53 @@ no-tasks = true
...
+| `--pprof-disabled` | `INFLUXD_PPROF_DISABLED` | `pprof-disabled`  |
+
+###### influxd flag
+```sh
+influxd --pprof-disabled
+```
+
+###### Environment variable
+```sh
+export INFLUXD_NO_TASKS=true
```
Comment (sanderson) [inline content/influxdb/v2.0/reference/config-options.md:840]:
```suggestion
export INFLUXD_PPROF_DISABLED=true
```

**Review by sanderson — CHANGES_REQUESTED** [review]

File: content/influxdb/v2.0/reference/config-options.md (hunk 3 above)
Comment (pierwill) [reply content/influxdb/v2.0/reference/config-options.md:840]: Good catch 😬

**Review by sanderson — APPROVED** [review]

**Pull request merged**